compiler-core/src/language_server/engine.rs

```diff
@@ -165,13 +161,13 @@ where
             let (uri, line_numbers) = match location.module {
                 None => (params.text_document.uri, &line_numbers),
                 Some(name) => {
-                    let module = match this.compiler.get_source(name) {
-                        Some(module) => module,
-                        // TODO: support goto definition for functions defined in
-                        // different packages. Currently it is not possible as the
-                        // required LineNumbers and source file path information is
-                        // not stored in the module metadata.
-                        None => return Ok(None),
+                    let module = match (
+                        this.compiler.get_source(name),
+                        this.compiler.get_dependency_source(name),
+                    ) {
+                        (Some(module), _) => module,
+                        (_, Some(module)) => module,
+                        _ => return Ok(None),
```

Comment:
Can you write this as `get_source(name).or_else(|_| get_dep_source(name))` please